.envrc

```diff
@@ -159,57 +144,59 @@ done
 
 ### Git ###
 
-info "Configuring git..."
+debug "Configuring git..."
 
 make setup-git-config
 
 ### Python ###
 
 venv_name=".venv"
 
-info "Activating virtualenv..."
-
 if [ ! -f "${venv_name}/bin/activate" ]; then
-    info "You don't seem to have a virtualenv."
-    # If the version doesn't match the contents of .python-version it will fail
-    [[ $(pyenv version) ]] ||
-        die "pyenv does not have the right Python version installed. Please run:
-    make setup-pyenv && python3.6 -m venv ${venv_name}"
-    advice_init_venv "$venv_name"
+    warn "You are missing a Python virtualenv and we ${bold}need${reset} to run a bootstrapping script (it can take a few minutes)"
+    echo -e "\nContinue (y/N)?"
+    read -r resp
+    case "$resp" in
+    y | Y) echo "Okay, let's do this." ;;
+    *)
+        die "Aborted!"
+        ;;
+    esac
+    # This is time consuming but it has to be done
+    source ./scripts/bootstrap-py3-venv
 fi
 
 # The user might be cd'ing into sentry with another non-direnv managed
 # (in that it would be automatically deactivated) virtualenv active.
 deactivate 2>/dev/null || true
 
+# shellcheck disable=SC1091
 source "${venv_name}/bin/activate"
 
 # XXX: ideally, direnv is able to export PS1 as modified by sourcing venvs
 #      but we'd have to patch direnv, and ".venv" isn't descriptive anyways
 unset PS1
 
-info "Ensuring proper virtualenv..."
-${PWD}/scripts/ensure-venv.sh
+debug "Ensuring proper virtualenv..."
+"${PWD}/scripts/ensure-venv.sh"
 
 if ! require sentry; then
-    info "Your virtualenv is activated, but sentry doesn't seem to be installed."
-    # XXX: if direnv fails, the venv won't be activated outside of direnv execution...
-    # So, it is critical that make install-py-dev is guarded by scripts/ensure-venv.
-    advice_install_sentry
+    warn "Your virtualenv is activated, but sentry doesn't seem to be installed."
+    commands_to_run+=("make install-py-dev")
 fi
 
 ### pre-commit ###
 
-info "Checking pre-commit..."
+debug "Checking pre-commit..."
 
-if ! require pre-commit; then
```

Comment:
Moving my comments here, but no longer using `require` is a regression in performance of direnv. It's noticeably slower now that it has to execute pre-commit

Reply:
Thanks for pointing that out. I will handle it in the PR I've opened.